### DISCUSSION TITLE
Feature request: Load file list asynchronously / cancel visiting directory

### DISCUSSION THREAD

**Discussion by andraaspar** [doublecmd/doublecmd#1168] [Suggestions]
Hi,

I am a long time day-to-day user of DoubleCmd and would like to thank you for developing it! It is awesome!

I would like to propose a new feature:

Loading a file list in a panel should happen asynchronously. This means, when I'm entering a folder with lots of files, or a folder at a slow network location, I should be able to work with the other pane while it loads. If it's possible, I would also like to be able to cancel visiting a location that loads for too long.

Currently DoubleCmd appears to hang while it waits for the file list to load. If the location is unavailable, DoubleCmd times out after a while, but it takes a long time (around a minute?) until it shows an error and I can get back to work. Being able to cancel the operation would help with that.

Thanks for considering!

**Comment by alexx2000** [comment]
Hi,

DC already work in this way when  `Load file list in separate thread` is [enabled](https://doublecmd.github.io/doc/en/configuration.html#ConfigViewEx). But it not works always.

**Comment by andraaspar** [comment]
Thank you for pointing that out, that option was already enabled on my computer.

I observed the issue more closely and you are right, it works in most cases. So maybe this is a bug report rather than a feature request. 🥲 

There is one case where it does not: when the network drive is unavailable. In that case, the entire application freezes, and I can see (Not Responding) in the title bar. It takes 20 seconds to 1 minute to recover from that state. Then I get a ‘Disk is not available’ message.

Also, if DoubleCmd is shut down with a network drive open in one of the panels, on the next startup, if the drive is unavailable, it will display the startup logo screen for 1:30 before it can launch. Upon launch it shows the network drive as empty. No error message.

Sorry to be complaining. It's just that because of my workflow I seem to be hitting this quite often recently.

Thank you for taking the time to look into this. I hope this helps. I am on:

```
Double Commander
Version: 1.0.6 beta
Revision: 121
Commit: e3012a9d0
Build date: 2022/06/04
Lazarus: 2.2.3.0
FPC: 3.2.2
Platform: x86_64-Win64-win32/win64
OS version: Windows 11 21H2 x86_64
```
